src/main/java/edu/kis/powp/jobs2d/command/manager/CommandManager.java

```diff
@@ -32,18 +33,23 @@ public synchronized void setCurrentCommand(DriverCommand commandList) {
      * @param commandList list of commands representing a compound command.
      * @param name        name of the command.
      */
-    public synchronized void setCurrentCommand(List<DriverCommand> commandList, String name) {
+    public synchronized void setCurrentCommand(List<IDriverCommand> commandList, String name) {
         setCurrentCommand(new ICompoundCommand() {
 
-            List<DriverCommand> driverCommands = commandList;
+            List<IDriverCommand> driverCommands = commandList;
 
             @Override
             public void execute(Job2dDriver driver) {
                 driverCommands.forEach((c) -> c.execute(driver));
             }
 
             @Override
-            public Iterator<DriverCommand> iterator() {
+            public void accept(CommandCounterVisitor commandCounterVisitor) {
+                driverCommands.forEach((c) -> c.accept(commandCounterVisitor));
```

Comment:
iteration should be done inside the visitor: public void visit(ICompoundCommand command) { ... }